--- a/Source/cmFindLibraryCommand.cxx
+++ b/Source/cmFindLibraryCommand.cxx
@@ -158,6 +158,7 @@
 
 void cmFindLibraryCommand::AddLib64Paths()
 {  
+#if 0 /* should not be done on Debian, just make it do nothing */
   if(!this->Makefile->GetLocalGenerator()->GetGlobalGenerator()->
      GetLanguageEnabled("C"))
     {
@@ -205,6 +206,7 @@
     {
     this->SearchPaths = path64;
     }
+#endif
 }
 
 //----------------------------------------------------------------------------
