--- a/Modules/FindPerlLibs.cmake
+++ b/Modules/FindPerlLibs.cmake
@@ -8,6 +8,7 @@
 #
 
 SET(PERL_POSSIBLE_INCLUDE_PATHS
+  /usr/lib/perl/5.10.0/CORE
   /usr/lib/perl/5.8.3/CORE
   /usr/lib/perl/5.8.2/CORE
   /usr/lib/perl/5.8.1/CORE
@@ -48,7 +49,7 @@
 ENDIF(PERL_EXECUTABLE)
 
 FIND_LIBRARY(PERL_LIBRARY
-  NAMES ${PERL_POSSIBLE_LIBRARY_NAME} perl5.8.0
+  NAMES ${PERL_POSSIBLE_LIBRARY_NAME} perl5.8.0 perl
   PATHS ${PERL_POSSIBLE_LIB_PATHS}
   )
 
