Subject: Modules/FindJNI.cmake: support VTK 5.4
 Backport Modules/FindJNI.cmake from CMake 2.8 CVS.
Origin: upstream, http://public.kitware.com/cgi-bin/viewcvs.cgi/Modules/FindVTK.cmake?root=CMake&r1=1.29&r2=1.31&view=patch
Bug: http://public.kitware.com/Bug/view.php?id=8804
Bug-Debian: http://bugs.debian.org/546030

--- a/Modules/FindVTK.cmake
+++ b/Modules/FindVTK.cmake
@@ -27,6 +27,19 @@
 #                 instead.
 #
 
+#=============================================================================
+# Copyright 2001-2009 Kitware, Inc.
+#
+# Distributed under the OSI-approved BSD License (the "License");
+# see accompanying file Copyright.txt for details.
+#
+# This software is distributed WITHOUT ANY WARRANTY; without even the
+# implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the License for more information.
+#=============================================================================
+# (To distributed this file outside of CMake, substitute the full
+#  License text for the above reference.)
+
 # Construct consitent error messages for use below.
 SET(VTK_DIR_DESCRIPTION "directory containing VTKConfig.cmake.  This is either the root of the build tree, or PREFIX/lib/vtk for an installation.  For VTK 4.0, this is the location of UseVTK.cmake.  This is either the root of the build tree or PREFIX/include/vtk for an installation.")
 SET(VTK_DIR_MESSAGE "VTK not found.  Set the VTK_DIR cmake cache entry to the ${VTK_DIR_DESCRIPTION}")
@@ -45,6 +58,7 @@
   SET(VTK_DIR_SEARCH "")
   FOREACH(dir ${VTK_DIR_SEARCH2})
     SET(VTK_DIR_SEARCH ${VTK_DIR_SEARCH}
+      ${dir}/../lib/vtk-5.4
       ${dir}/../lib/vtk-5.2
       ${dir}/../lib/vtk-5.1
       ${dir}/../lib/vtk-5.0
