--- a/Modules/FindBoost.cmake
+++ b/Modules/FindBoost.cmake
@@ -66,7 +66,7 @@
 #
 # SET(Boost_ADDITIONAL_VERSIONS "0.99" "0.99.0" "1.78" "1.78.0")
 #
-# ============================================================================
+# ===================================== ============= ========================
 #
 # Variables used by this module, they can change the default behaviour and
 # need to be set before calling find_package:
@@ -147,7 +147,7 @@
 #                                       "component" (includes debug and optimized keywords
 #                                       when needed).
 #
-# =====================================================================
+# ===================================== ============= =================
 #
 #
 #  Copyright (c) 2006-2008 Andreas Schneider <mail@cynapses.org>
--- a/Modules/FindRTI.cmake
+++ b/Modules/FindRTI.cmake
@@ -5,7 +5,7 @@
 # RTI is a simulation infrastructure standartized by IEEE and SISO. It has a
 # well defined C++ API that assures that simulation applications are
 # independent on a particular RTI implementation.
-# http://en.wikipedia.org/wiki/Run-Time_Infrastructure_(simulation)
+#  http://en.wikipedia.org/wiki/Run-Time_Infrastructure_(simulation)
 #
 # This code sets the following variables:
 #  RTI_INCLUDE_DIR = the directory where RTI includes file are found
